File: module2/25-2-Edittext,Validation,SPlashScreen,AlertDialog/MainActivity2.java
package com.example.splashscreen;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

public class MainActivity2 extends AppCompatActivity {
    ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        imageView=findViewById(R.id.img1);

        imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url="https://www.javatpoint.com/android-tutorial";
                Intent i=new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);
            }
        });
    }
  /*  @Override
    public void onBackPressed()
    {
        AlertDialog.Builder alert=new AlertDialog.Builder(MainActivity2.this);
        alert.setTitle("Are you sure you want to exit?");
        DialogInterface.OnClickListener{
            @Override
                    public void OnClick(DialogInterface dialogInterface,int i)
        {
            finishActivity();
        }
    }
    }*/
}

File: module2/21-3-Material Design/MainActivity2.kt
package com.example.my_fragment

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.my_fragment.databinding.ActivityMain2Binding

class MainActivity2 : AppCompatActivity()
{
    private lateinit var binding: ActivityMain2Binding

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        binding = ActivityMain2Binding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.btn1.setOnClickListener {
            var f=FirstFragment()
            var fm=supportFragmentManager
            var ft=fm.beginTransaction()
            ft.replace(R.id.frmid,f).commit()
        }
    }
}